packages/freezed/lib/src/freezed_generator.dart

```diff
@@ -547,6 +562,30 @@ extension on Element {
   }
 }
 
+extension on ConstructorElement {
+  String unionValue(FreezedUnionCase unionCase) {
+    final annotation = const TypeChecker.fromRuntime(FreezedUnionValue)
+        .firstAnnotationOf(this, throwOnUnresolved: false);
+    if (annotation != null) {
+      return annotation.getField('value').toStringValue();
+    }
+
+    final constructorName = isDefaultConstructor(this) ? 'default' : name;
+    switch (unionCase) {
+      case FreezedUnionCase.none:
+        return constructorName;
+      case FreezedUnionCase.kebab:
+        return kebabCase(constructorName);
+      case FreezedUnionCase.pascal:
+        return pascalCase(constructorName);
+      case FreezedUnionCase.snake:
+        return snakeCase(constructorName);
+    }
+
+    throw ArgumentError('Unknown unionCase: $unionCase}');
```

Comment:
Instead do:

```dart
case ...:
default:
  throw FallthroughError();
```

Reply:
I'm ok to change it, but personally I don't like `default` in enum switch, because without it, the analyser will at least give a warning if another case is added to enum.